lib/kbuildbarn/exec/bb_clientd.go

```diff
@@ -75,6 +75,25 @@ func MaybeStartClient(o *ClientOptions) (*Client, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error starting bb_clientd: %w", err)
 	}
+
+	// Wait for a directory under the mount dir to appear, so that we know
+	// bb_clientd is ready, up until `timeout`
+	stopTime := time.Now().Add(timeout)
+	for {
+		time.Sleep(100*time.Millisecond)
+		if time.Now().After(stopTime) {
+			return nil, fmt.Errorf("failed to find mount directory ready after %v", timeout)
```

Comment:
when you return here for errors, you are leaking the process? eg, next run of kbuildbarn may fail / be unreliable / be flaky, as a background bb_clientd may (or may not) still be running (and trying, or not to be using the directory)?

I would recommend some form of defer to kill the process if this function returns with error. Ideally with a waitpid() equivalent, so you can log if it failed in any way (right now failure is silently discarded?)

Reply:
Done; now using named returns, and after the process is created a defer checks the returned error and runs the existing Shutdown codepath to clean up if necessary.